bzip2.rs

```diff
@@ -903,6 +903,31 @@ unsafe fn copy_filename(to: *mut c_char, from: &str) {
     *to.wrapping_add(FILE_NAME_LEN - 10) = '\0' as i32 as c_char;
 }
 
+fn fopen_output_safely_rust(name: impl AsRef<Path>) -> *mut FILE {
+    use std::os::fd::AsRawFd;
+    use std::os::unix::fs::OpenOptionsExt;
+
+    let mut opts = std::fs::File::options();
+
+    opts.write(true).create_new(true);
+
+    opts.mode(libc::S_IWUSR | libc::S_IRUSR);
+
+    let Ok(file) = opts.open(name) else {
+        return std::ptr::null_mut::<FILE>();
+    };
+
+    let mode = b"wb\0".as_ptr().cast::<c_char>();
+    let fp = unsafe { fdopen(file.as_raw_fd(), mode) };
+    if fp.is_null() {
+        drop(file)
+    } else {
+        core::mem::forget(file)
```

Comment:
I feel like it would be nicer to use `.into_raw_fd()` above rather than `forget(file)` here and then use `close(fd)` in the `fp.is_null()` branch.

Reply:
for now, probably yes